protobuf/lib/src/protobuf/pb_list.dart

```diff
@@ -6,310 +6,216 @@ part of protobuf;
-  bool any(bool Function(E element) test) => _wrappedList.any(test);
-
-  @override
-  List<E> toList({bool growable = true}) =>
-      _wrappedList.toList(growable: growable);
-
-  @override
-  Set<E> toSet() => _wrappedList.toSet();
-
-  @override
-  bool get isEmpty => _wrappedList.isEmpty;
-
-  @override
-  bool get isNotEmpty => _wrappedList.isNotEmpty;
-
-  @override
-  Iterable<E> take(int count) => _wrappedList.take(count);
-
-  @override
-  Iterable<E> takeWhile(bool Function(E value) test) =>
-      _wrappedList.takeWhile(test);
-
-  @override
-  Iterable<E> skip(int count) => _wrappedList.skip(count);
-
-  @override
-  Iterable<E> skipWhile(bool Function(E value) test) =>
-      _wrappedList.skipWhile(test);
-
-  @override
-  E get first => _wrappedList.first;
-
-  @override
-  E get last => _wrappedList.last;
-
-  @override
-  E get single => _wrappedList.single;
-
-  @override
-  E firstWhere(bool Function(E element) test, {E Function()? orElse}) =>
-      _wrappedList.firstWhere(test, orElse: orElse);
-
-  @override
-  E lastWhere(bool Function(E element) test, {E Function()? orElse}) =>
-      _wrappedList.lastWhere(test, orElse: orElse);
-
-  @override
-  E elementAt(int index) => _wrappedList.elementAt(index);
-
-  @override
-  String toString() => _wrappedList.toString();
-
   @override
   E operator [](int index) => _wrappedList[index];
 
-  @override
-  int get length => _wrappedList.length;
-
-  // TODO(jakobr): E instead of Object once dart-lang/sdk#31311 is fixed.
-  @override
-  int indexOf(Object? element, [int start = 0]) =>
-      _wrappedList.indexOf(element as E, start);
-
-  // TODO(jakobr): E instead of Object once dart-lang/sdk#31311 is fixed.
-  @override
-  int lastIndexOf(Object? element, [int? start]) =>
-      _wrappedList.lastIndexOf(element as E, start);
-
-  @override
-  List<E> sublist(int start, [int? end]) => _wrappedList.sublist(start, end);
-
-  @override
-  Iterable<E> getRange(int start, int end) => _wrappedList.getRange(start, end);
-
-  @override
-  Map<int, E> asMap() => _wrappedList.asMap();
-
   @override
   void operator []=(int index, E value) {
-    check(value);
+    if (_isReadOnly) {
+      throw UnsupportedError('Setting field of a read-only list');
+    }
+    _check(value);
     _wrappedList[index] = value;
   }
 
-  /// Unsupported -- violated non-null constraint imposed by protobufs.
-  ///
-  /// Changes the length of the list. If [newLength] is greater than the current
-  /// [length], entries are initialized to [:null:]. Throws an
-  /// [UnsupportedError] if the list is not extendable.
   @override
-  set length(int newLength) {
-    if (newLength > length) {
-      throw UnsupportedError('Extending protobuf lists is not supported');
+  bool operator ==(other) => (other is PbList) && _areListsEqual(other, this);
+
+  @override
+  int get hashCode => _HashUtils._hashObjects(_wrappedList);
+
+  void freeze() {
+    _isReadOnly = true;
+    if (_isGroupOrMessage(_elementType)) {
```

Comment:
This part is buggy. It was copied from `PbMap` which doesn't allow `map` or `repeated` values.

Reply:
As far as I can tell You can not have a repeated map: https://developers.google.com/protocol-buffers/docs/proto3#maps

> Map fields cannot be repeated.

I also don't think you can have a `repeated` of a `repeated`.

So maybe this is good enough.

Reply:
You are right! I added a line of comment there to clarify.